core/src/main/scala/feral/IOSetup.scala

```diff
@@ -20,24 +20,28 @@ import cats.effect.unsafe.IORuntime
 import cats.effect.kernel.Resource
 import cats.effect.IO
 import cats.effect.kernel.Deferred
+import cats.effect.std.Dispatcher
 
 private[feral] trait IOSetup {
+  private def runtime =
+    IORuntime.global
 
-  protected def runtime: IORuntime = IORuntime.global
+  protected def dispatcher: Dispatcher[IO] =
+    Dispatcher.sequential[IO](await = true).allocated.map(_._1).unsafeRunSync()(runtime)
 
   protected type Setup
   protected def setup: Resource[IO, Setup] = Resource.pure(null.asInstanceOf[Setup])
 
   private[feral] final lazy val setupMemo: IO[Setup] = {
     val deferred = Deferred.unsafe[IO, Either[Throwable, Setup]]
```

Comment:
Since this is a one time setup, there's not much reason to run it on the `Dispatcher`. In fact, we could setup the `Dispatcher` and the memo in the same unsafe call.